scripts/tests/chiptest/linux.py

```diff
@@ -197,6 +199,9 @@ def terminate(self):
         for command in self.COMMANDS_TERMINATE:
             self.run(command)
 
+    def wrap_in_namespace(self, subprocess: Subprocess):
+        return subprocess.wrap_with(("ip", "netns", "exec", "{}-{}".format(subprocess.kind, self.index)))
```

Comment:
Running process in a namespace this way is very Linux specific, I think. I can imagine a more complicated method on some other platform (if there are any...). More generic way would be to provide a dedicated "executor" which will start the subprocess in the correct namespace (the `ip netns exec` should be an implementation detail within the "LinuxNamespaceSubprocess" class). Then, you could use that "executor" in the `runner.py::Runner.RunSubprocess()` instead of `subprocess.Popen`.